modules/ximgproc/src/seeds.cpp

```diff
@@ -1072,14 +1074,14 @@ int SuperpixelSEEDSImpl::fourbythree(int x, int y, int label)
     count += (labels[(y - 1) * width + x] == label);
     count += (labels[(y - 1) * width + x + 1] == label);
 
-    count += (labels[y * width + x - 1] == label);
+    count += (labels[(y + 2) * width + x] == label);
     count += (labels[y * width + x + 2] == label);
 
     count += (labels[(y + 1) * width + x - 1] == label);
     count += (labels[(y + 1) * width + x + 2] == label);
 
     count += (labels[(y + 2) * width + x - 1] == label);
-    count += (labels[(y + 2) * width + x] == label);
+    count += (labels[y * width + x - 1] == label);
```

Comment:
It seems that you've replaced this line with 1077 one. Is it necessary? I'm asking because these lines are following one by another without  side effects (seems to the code)